CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -123,6 +124,7 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
         if(pupCol->refAt(iPF).key() != *itPho) continue;
         pWeight = weight_;
         if(!useValueMap_ && itPF->pt() != 0) pWeight = pWeight*(phoCands[iPho]->pt()/itPF->pt());
+	if(!useValueMap_ && itPF->pt() == 0) pVec.SetPxPyPzE(phoCands[iPho]->px()*pWeight,phoCands[iPho]->py()*pWeight,phoCands[iPho]->pz()*pWeight,phoCands[iPho]->energy()*pWeight);
```

Comment:
Removing this line is not an option. This makes the MET performance
significantly worse.

On Wed, Sep 7, 2016 at 3:31 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In CommonTools/PileupAlgos/plugins/PuppiPhoton.cc
> <https://github.com/cms-sw/cmssw/pull/15562#discussion_r77821602>:
>
> > @@ -123,6 +124,7 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
> >          if(pupCol->refAt(iPF).key() != *itPho) continue;
> >          pWeight = weight_;
> >          if(!useValueMap_ && itPF->pt() != 0) pWeight = pWeight*(phoCands[iPho]->pt()/itPF->pt());
> > +	if(!useValueMap_ && itPF->pt() == 0) pVec.SetPxPyPzE(phoCands[iPho]->px()*pWeight,phoCands[iPho]->py()*pWeight,phoCands[iPho]->pz()*pWeight,phoCands[iPho]->energy()*pWeight);
>
> (workflow running AOD->miniAOD) looking event by event at saved MET vs
> sum(puppiMerged), it looks like this line is executed every event for every
> photon in the collection.
> In the first 10 events I see roughly "-sum(puppiMerged.px) -
> sum(slimmedPhotons.px) ~ pfMetPuppi.px"
>
> after removing this line, in most events "-sum(puppiMerged.px) =
> pfMetPuppi.px", except for a few (maybe one in 10) where there is
> difference.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15562/files/691379a0f452cf21724f83c3978bf13682245d8b#r77821602>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE7V4dWFMJtj4_aJNc_C8xPZ80uqJU0lks5qnrymgaJpZM4JqyuB>
> .
>